packages/core/admin/admin/src/content-manager/components/BlocksInput/Blocks/Link.tsx

```diff
@@ -51,35 +40,45 @@ const LinkContent = React.forwardRef<HTMLAnchorElement, LinkContentProps>(
     const [popoverOpen, setPopoverOpen] = React.useState(
       editor.lastInsertedLinkPath ? Path.equals(path, editor.lastInsertedLinkPath) : false
     );
-    const [isEditing, setIsEditing] = React.useState(link.url === '');
     const linkRef = React.useRef<HTMLAnchorElement>(null!);
     const elementText = link.children.map((child) => child.text).join('');
     const [linkText, setLinkText] = React.useState(elementText);
     const [linkUrl, setLinkUrl] = React.useState(link.url);
+    const linkInputRef = React.useRef<HTMLInputElement>(null);
+    const [showRemoveButton, setShowRemoveButton] = React.useState(false);
+    const [invalidLinkError, setInvalidLinkError] = React.useState('');
 
     const handleOpenEditPopover: React.MouseEventHandler<HTMLAnchorElement> = (e) => {
       e.preventDefault();
       setPopoverOpen(true);
+      setShowRemoveButton(true);
```

Comment:
I'm guessing the only case where this won't be true is if you create a new link via the toolbar. But I can't find how the popover would be opened in that case. Could you explain?

Reply:
Its the same popover we show on clicking new link via toolbar. So on clicking we are inserting a new link and save its path, if current path is same as last inserted link path we show the popover. @Feranchz can explain more if its not clear.